frontend/javascripts/oxalis/model/sagas/min_cut_saga.js

```diff
@@ -0,0 +1,722 @@
+
+    if (V3.equals(currVoxel, seedB)) {
+      foundTarget = true;
+      break;
+    }
+
+    const neighbors = getNeighborsFromBitMask(edgeBuffer[currVoxelIdx]).outgoing;
+    for (const neighbor of neighbors) {
+      const neighborPos = V3.add(currVoxel, neighbor);
+      const neighborIdx = getNeighborIdx(neighbor);
+
+      if (distanceField[ll(neighborPos)] === 0) {
+        queue.push({ voxel: neighborPos, distance: distance + 1, usedEdgeIdx: neighborIdx });
+      }
+    }
+  }
+
+  return { distanceField, directionField, foundTarget };
+}
+
+function removeShortestPath(
+  distanceField: Uint16Array,
+  directionField: Uint8Array,
+  seedA,
+  seedB,
+  ll,
+  size,
+  edgeBuffer,
+  minDistToSeed,
+) {
+  // Extract shortest path from seedB to seedA and remove edges which
+  // belong to that path.
+  const path = [];
+  let foundSeed = false;
+  const voxelStack = [seedB];
+  const maxDistance = distanceField[ll(seedB)];
+  let removedEdgeCount = 0;
+
+  while (voxelStack.length > 0) {
+    const currentVoxel = voxelStack.pop();
+    const currentDistance = distanceField[ll(currentVoxel)];
+
+    if (V3.equals(currentVoxel, seedA)) {
+      foundSeed = true;
+      break;
+    }
+
+    const originallyUsedEdgeId = directionField[ll(currentVoxel)];
+    if (originallyUsedEdgeId >= NEIGHBOR_LOOKUP.length) {
+      throw new Error("Could not look up used edge in directionField");
+    }
+    const neighborIdx = invertNeighborIdx(originallyUsedEdgeId);
+    const neighbor = NEIGHBOR_LOOKUP[neighborIdx];
+
+    const neighborPos = V3.add(currentVoxel, neighbor);
+
+    if (isPositionOutside(neighborPos, size)) {
+      throw new Error("Neighbor is outside of volume?");
+    }
+
+    if (!(distanceField[ll(neighborPos)] < currentDistance && distanceField[ll(neighborPos)] > 0)) {
+      throw new Error("Direction points towards a higher distance?");
+    }
+
+    const currDist = distanceField[ll(neighborPos)];
+    const distToSeed = Math.min(currDist, maxDistance - currDist);
+
+    // We don't want to remove voxels which are close to the seeds.
+    // See explanation for MIN_DIST_TO_SEED for details.
+    if (distToSeed > minDistToSeed) {
+      removedEdgeCount++;
+      path.unshift(neighborPos);
+
+      // Remove ingoing edge
+      removeIngoingEdge(edgeBuffer, ll(currentVoxel), neighborIdx);
+
+      // Remove outgoing edge
+      const invertedNeighborIdx = invertNeighborIdx(neighborIdx);
+      removeOutgoingEdge(edgeBuffer, ll(neighborPos), invertedNeighborIdx);
+    }
+
+    voxelStack.push(neighborPos);
+  }
+
+  return { path, foundSeed, removedEdgeCount };
+}
+
+function traverseResidualsField(
+  boundingBoxTarget: BoundingBox,
+  seedA: Vector3,
+  ll,
+  edgeBuffer: Uint16Array,
+) {
+  // Perform a breadth-first search from seedA and return
+  // which voxels were visited (visitedField). This represents
+  // the graph component for seedA.
+  const visitedField = new Uint16Array(boundingBoxTarget.getVolume());
```

Comment:
Since the field only holds 0s and 1s, a `Uint8Array` is sufficient.
```suggestion
  const visitedField = new Uint8Array(boundingBoxTarget.getVolume());
```